test/framework/easyconfig.py

```diff
@@ -1278,6 +1278,17 @@ def test_constant_doc(self):
         ]
         self.assertEqual(len(doc.split('\n')), sum([len(temps)] + [len(x) for x in temps]))
 
+    def test_constants_import(self):
+        """Test importing constants works"""
+        # Sanity check that importing an EC constant works as-if using EASYCONFIG_CONSTANTS
+        from easybuild.framework.easyconfig.constants import SYSTEM
+        self.assertEqual(SYSTEM, easyconfig.constants.EASYCONFIG_CONSTANTS['SYSTEM'][0])
```

Comment:
For `SYSTEM`, I would make this a hard check, rather than relying on `EASYCONFIG_CONSTANTS`:

```python
self.assertEqual(SYSTEM, {'name': 'system', 'version': 'system'})
```

Reply:
But that is not what is tested here. Only importing the constants should be tested here and that means that doing such an explicit import is the same as accessing the `EASYCONFIG_CONSTANTS` list, see the comment.

Reply:
I'm not saying you should not do the explicit import, you should.

I'm saying you should compare the value of `SYSTEM` directly to a hardcoded dict here, rather than relying on `EASYCONFIG_CONSTANTS`, otherwise the tests would still pass when the value for `SYSTEM` in `EASYCONFIG_CONSTANTS` in changed in the code.

Reply:
> otherwise the tests would still pass when the value for `SYSTEM` in `EASYCONFIG_CONSTANTS` in changed in the code.

Erm, This is exactly what you want, don't you?

- The value of `SYSTEM` in ECs is that of `EASYCONFIG_CONSTANTS`
- Previously you would need to use `easyconfig.constants.EASYCONFIG_CONSTANTS['SYSTEM'][0]` in hooks etc.
- With this PR you can instead use `from easybuild.framework.easyconfig.constants import SYSTEM` to the same effect
- Hence the test checks that the imported value matches the value in `EASYCONFIG_CONSTANTS` to verify it is indeed the same effect and has a matching comment

So I don't understand why the test shouldn't pass when `easybuild.framework.easyconfig.constants.SYSTEM == easyconfig.constants.EASYCONFIG_CONSTANTS['SYSTEM'][0]`. Could you explain your thought here?

Additionally I really dislike using hardcoded values as this results in unrelated tests failing. This was e.g. the case where a spelling fix to a doc caused another test to fail: https://github.com/easybuilders/easybuild-framework/pull/3219 -> https://github.com/easybuilders/easybuild-framework/pull/4158 So if some previously run other test for any reason changed the `EASYCONFIG_CONSTANTS` (which shouldn't happen, but I guess is what you had in mind) this test still passing is IMO actually a good thing.

Reply:
I consider a change to the value of a constant a breaking change, so tests should fail because of it.

Although no code *should* be relying on the value of the constant, it's not unlikely that some stuff does, and hence the value of the constant should never be changed.
Hardcoding the expected value in the tests is a way of catching that.

Reply:
> I consider a change to the value of a constant a breaking change, so tests should fail because of it.

Ok understood. But I wouldn't do that here as that is not the purpose of the test: "importing an EC constant works as-if using EASYCONFIG_CONSTANTS".

I split the test into 2 where the 2nd ensures exactly that and the first is more like what a user would do and uses the hard-coded value. I'm still not a fan of doing that for the mentioned reason and would rather have dedicated tests to catch constant changes that shouldn't change or just say that the purpose of the constant is that the value does not matter so anything relying on it is a bug anyway.